### PR TITLE
feat: Support for Python 'setup.cfg' files

### LINKED ISSUE #2734: Support python setup.cfg
**What would you like Renovate to be able to do?**

Parse and update Python `setup.cfg` files.

**Describe the solution you'd like**

Same as `pip_requirements` but make sure we don't do any auto pinning.

**Describe alternatives you've considered**

Some users may still use `setup.py` (#2681)

**Additional context**

See https://github.com/renovatebot/renovate/issues/2681#issuecomment-434488054 from @keimlink

### PR DESCRIPTION
**Pull request by zharinov**
Closes #2734

I tried to use `ini` library, but it was unable to parse multi-line dependency lists, that's why we're using line-by-line parser here.

### CODE REVIEW

File: lib/manager/setup-cfg/range.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { RangeStrategy } from '../../types';
+import { RangeConfig } from '../common';
+
+export function getRangeStrategy(config: RangeConfig): RangeStrategy {
+  return 'bump';
```
[the remark this answers is not in the document]
Comment (zharinov) [reply lib/manager/setup-cfg/range.ts:6]: If this is okay, then ready for review

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/manager/setup-cfg/range.ts:6]: I don't see any reason to default to bumping. If the range is satisfied by the latest then we don't need to update it. Let's return to `replace`. You can test with out-of-range range updates or pinned ones if you want?

**Review by rarkins — CHANGES_REQUESTED** [review]
Don't bump by default

File: lib/manager/setup-cfg/range.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zharinov) [reply lib/manager/setup-cfg/range.ts:6]: Ah, I see `~=2.1` is equivalent to `>= 2.1, ==2.*`, so even `2.8.1` is okay. Thus, the only 3 updates with `replace` strategy is okay for test repo.

[the remark this answers is not in the document]
Comment (zharinov) [reply lib/manager/setup-cfg/range.ts]: [Pinned deps](https://github.com/zharinov/renovate-setup-cfg/pull/44), not sure should we preserve `==` there or not

File: lib/manager/setup-cfg/extract.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export function extractPackageFile(
+  content: string
+): Result<PackageFile | null> {
+  let sectionName = null;
+  let sectionRecord = null;
+
+  const deps: PackageDependency[] = [];
+  content
```
Comment (rarkins) [inline lib/manager/setup-cfg/extract.ts:61]: Are you sure you don't want to use a .ini parser for this?

Comment (zharinov) [reply lib/manager/setup-cfg/extract.ts:61]: I tried it at first, but for some reason it returned `undefined` for multiline `install_requres`

**Comment by augi** [comment]
Hello, what is please name of the _manager_ to be enable to get this feature? I'm speaking about managers as described here: lib/manager/setup-cfg/extract.ts
Thanks!

**Comment by viceice** [comment]
It's enabled by default, if you use the app, you need to wait for next update

**Comment by augi** [comment]
I'm using the Docker image and also the `enabledManagers` option so I have to know exact name of this new manager.
Or is it just a new feature of an existing manager?

Thanks!

**Comment by viceice** [comment]
see https://docs.renovatebot.com/modules/manager/ for all managers

**Comment by augi** [comment]
Thank you! That is exactly why I'm asking here - it looks like [this documentation](https://docs.renovatebot.com/configuration-options/#enabledmanagers) seems not to be updated because `setup-cfg` is missing there, correct?

**Comment by rarkins** [comment]
Yeah, maybe that field isn't auto generated. Perhaps we should instead link to the manager modules URL because that one *is* auto generated every build.

**Comment by augi** [comment]
That would be great! 👍

**Pull request merged**